lib/order.rb

```diff
@@ -1,18 +1,80 @@
+require 'awesome_print'
+require 'csv'
+
 module Grocery
   class Order
-    attr_reader :id, :products
+    attr_reader :id
+    attr_accessor :order, :products, :gross_total
+
+    @@organized_orders = []
 
-    def initialize(id, products)
+    def initialize (id, products)
       @id = id
       @products = products
     end
 
     def total
-      # TODO: implement total
+      subtotal = 0
+      @products.each do |product, price|
+        subtotal += price
+      end
+      tax = subtotal * 0.075
+      @gross_total = (subtotal + tax).round(2)
+      return @gross_total
+    end
+    # Takes user input product, price and order id
+    # Allows user to add new order, or add product and its price to existing order
+    # Appends new order onto existing orders csv file
+    def add_product (product_name, product_price, add_id)
+      if add_id != "new"
+        add_id = add_id.to_i
+        add_products = self.find_by_id(add_id)
+        if !add_product == false
+          add_products.each do |product, price|
+            if add_product == product
+              puts "The product has already been added."
+              return false
+            else
+              add_products.store(product_name, product_price)
+              puts "#{product_name} has been added to #{add_id}"
+              return true
+            end
+          end
+        else
+          new_id = orders.length + 1
+          new_products = {product_name => product_price}
+          CSV.open("../support/orders.csv", 'a') do |orders_csv|
+            orders_csv << ["#{new_id}", "#{product_name}:#{product_price}"]
+          end
+          # return is better because allows for continuous updating of csv from multiple programs.
+          @@organized_orders << Order.new(new_id, new_products)
+        end
+      end
+    end
+
+    def self.find (find_id)
```

Comment:
This method should either test to see if `@@organized_orders` is nil and if so, then call `self.all`